src/env/default.rs

```diff
@@ -165,6 +172,33 @@ impl LogFd {
             Ok(())
         }
     }
+
+    #[allow(unused_variables)]
+    pub fn allocate_with_hole(&self, offset: usize, size: usize) -> IoResult<()> {
+        fail_point!("log_fd::falloc_with_hole::err", |_| {
+            Err(from_nix_error(nix::Error::EINVAL, "fp"))
+        });
+        #[cfg(target_os = "linux")]
+        {
+            fcntl::fallocate(
+                self.0,
+                // It'a special flag for users who want to `fallcate` a file with
+                // fill a `hole`(that is, zero) into the specified `offset + size`
```

Comment:
This is incorrect and I don't think we will need it. Punch hole will not allocate file blocks and certainly not fill zeros. It only marks the specified file range as uninitialized.

Reply:
My fault. It's a misleading annotation, and the func should be removed. Also, the explaination on flag `FALLOC_FL_PUNCH_HOLE` is totally incorrect.